themes/default/content/blog/pulumi-targets-dotnet-6/index.md

```diff
@@ -0,0 +1,178 @@
+---
+title: "Pulumi now targets .NET 6"
+authors: ["zaid-ajaj"]
+tags: ["dotnet", "csharp", "fsharp", "vb.net"]
+meta_desc: "Introducing a static-code analyzer for C# which provides instant feedback on common mistakes defining Pulumi resources"
+# meta_image: enhanced-static-code-analysis.png
+date: "2022-07-20"
+---
+
+In this blog post, we will talk about how Pulumi is now using [.NET 6](https://docs.microsoft.com/en-us/dotnet/core/whats-new/dotnet-6) as our default across the ecosystem. We will discuss the changes applied to templates, program structure and code generation. We also explain how Pulumi C# projects can benefit from latest features in .NET 6 and how it simplifies your programs overall. Let's dive in, shall we?
+
+<!--more-->
+
+### Templates
+
+All of our .NET templates have been updated which now use `net6.0` as the target framework. This means that projects created with `pulumi new` require at least the [.NET 6 SDK](https://dotnet.microsoft.com/en-us/download/dotnet/6.0) to build and run the projects. The templates also simplified the program structure quite a bit and not longer use `MyStack` classes to define resources in case of C# and VB.NET templates. F# templates didn't change that much since these didn't use the class approach to define infrastructure. 
+
+### Simplified program structure
+
+Let us examine a new C# template. Create a new project using `pulumi new aws-csharp` in a new directory and notice that there is no longer a `MyStack.cs` file, only `Program.cs`. Open the latter in an editor and you should see the following:
+
+```cs
+using Pulumi;
+using Pulumi.Aws.S3;
+using System.Collections.Generic;
+
+await Deployment.RunAsync(() =>
+{
+   // Create an AWS resource (S3 Bucket)
+   var bucket = new Bucket("my-bucket");
+   
+   // Export the name of the bucket
+   return new Dictionary<string, object?>
+   {
+      ["bucketName"] = bucket.Id
+   };
+});
+```
+
+This is how a stack looks like now with resource definitions and outputs. The [Get Started guide](https://www.pulumi.com/docs/get-started/) has been updated to follow this pattern as well.
+
+Targeting .NET 6 means that we can use features from C# 10 and here we see them in action: using [top-level statements](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/tutorials/top-level-statements) that greatly simplify the entry point of the project and reduce it down to a single function call to `Deployment.RunAsync`. Inside this function is where we define our resources and return outputs for the stack. Top-level statements also allow us to `await` asynchronous functions without having an explicit `Main(string[] args)` function or a `Program` class.
+
+The equivalent of the snippet above would have been this piece of code:
```

Comment:
I would be fine if we skipped the old code. Maybe link to a github hash that has it. Otherwise viewers may be confused if they aren't reading carefully.